unlock-wordpress-plugin/inc/classes/blocks/class-unlock-box-block.php

```diff
@@ -168,4 +179,196 @@ private function get_checkout_url( $attributes ) {
+
+        <div id="lock-network-inputs" style="display: none;">
+            <label for="unlock-pv2-network-label">Lock Networks:</label>
+            <select name="unlock-pv2-network-select" id="unlock-pv2-network-select" onchange="toggleLockIdInput()">
+                <option value="" <?php selected($meta_network, ""); ?>></option>
+                <?php foreach ($networkList['networks'] as $network) { ?>
+                    <option value="<?php echo esc_attr($network['network_name']); ?>" <?php selected($meta_network, $network['network_name']); ?>><?php echo esc_html($network['network_name']); ?></option>
+                <?php } ?>
+            </select>
+
+            <div id="lock-id-input" style="display: <?php echo ($meta_network ? "block" : "none"); ?>;">
+				<label for="unlock-pv2-lock-id-label">Lock Contract ID:</label>
+				<input type="text" name="unlock-pv2-lock-id-input" id="unlock-pv2-lock-id-input" value="<?php echo esc_attr($meta_id); ?>" />
+			</div>
+
+			<br>
+        	<div id="unlock-pv2-lock-setup-notification"></div>
+
+
+        </div>
+
+        <script>				
+
+			function toggleLockInputs() {
+                var lockInputs = document.getElementById("lock-network-inputs");
+                if (lockInputs.style.display === "none") {
+                    lockInputs.style.display = "block";
+                    document.getElementById("add-lock-button").innerHTML = "Remove Lock";
+                } else {
+                    lockInputs.style.display = "none";
+                    document.getElementById("add-lock-button").innerHTML = "Add Lock";
+                }
+            }
+
+            function toggleLockIdInput() {
+                var lockIdInput = document.getElementById("lock-id-input");
+                if (document.getElementById("unlock-pv2-network-select").value === "") {
+                    lockIdInput.style.display = "none";
+                } else {
+                    lockIdInput.style.display = "block";
+                }
+            }
+
+        </script>
+    <?php
+}
+  
+
+
+
+
+
+// Save lock custom meta data to the meta of post/page it was added
+function unlock_pv2_save_custom_meta($post_id) 
+{
+    if (isset($_POST['unlock_pv2_network_meta'])) {
+        
+        $network_input = sanitize_text_field( $_POST['unlock-pv2-network-select'] );
+        
+        if ($network_input) {
+
+            //Save lock network to the meta of specific post/page it was added
+            update_post_meta($post_id, 'unlock_pv2_network_meta', $network_input);
+
+            if (isset($_POST['unlock_pv2_id_meta'])) {
+
+                $id_input = sanitize_text_field( $_POST['unlock-pv2-lock-id-input'] );
+
+                if (!is_wp_error($id_input)) {
+
+                    //Save lock address id to the meta of specific post/page it was added
+                    update_post_meta($post_id, 'unlock_pv2_id_meta', $id_input);
```

Comment:
I think it would be better to use the same format to store the network(s) and lock address(es) than the one used for when the lock is applied to a block